geozero/src/wkb/wkb_reader.rs

```diff
@@ -178,17 +215,111 @@ fn read_gpkg_header<R: Read>(raw: &mut R) -> Result<WkbInfo> {
         has_z: ogc_info.has_z,
         has_m: ogc_info.has_m,
         srid: Some(srid),
+        is_compressed: false,
         envelope,
     };
     Ok(info)
 }
 
-// TODO: Spatialite https://www.gaia-gis.it/gaia-sins/BLOB-Geometry.html
+/// Spatialite WKB header.
+pub(crate) fn read_spatialite_header<R: Read>(raw: &mut R) -> Result<WkbInfo> {
+    let start = raw.ioread::<u8>()?;
+    if start != 0 {
+        return Err(GeozeroError::GeometryFormat);
+    }
+    let flags = raw.ioread::<u8>()?;
+    let endian = Endian::from(flags & 0b0000_0001 != 0);
+    let is_tinypoint = flags & 0b1000_0000 != 0;
+
+    let srid = match raw.ioread_with::<i32>(endian)? {
+        0 => None,
+        val => Some(val),
+    };
+
+    let info = if is_tinypoint {
+        let envelope = Vec::<f64>::new();
```

Comment:
many of these lines should probably be inlined for ease-of-reading. As long as the `raw.ioread_with...` is in the same order, it seems like it should be ok.  Also, i think `matches!(...)` is a bit hard to read -- perhaps a simple bool is better, e.g. `(type_id_dim == 2 || type_id_dim == 4)` ?  Or is this something clippy is having too much fun with?

Reply:
I personally find the `matches!(...)` macro easier on the eyes, but I'm fine with either - I don't think clippy has any problem with one solution or the other.